src/main/java/edu/kit/datamanager/metastore2/configuration/WebSecurityConfig.java

```diff
@@ -67,20 +68,21 @@ public class WebSecurityConfig {
   public WebSecurityConfig() {
   }
 
-  @Bean
-  public AuthenticationManager authenticationManager(AuthenticationConfiguration authenticationConfiguration) throws Exception {
-    return authenticationConfiguration.getAuthenticationManager();
-  }
-
-  @Bean
-  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-    AuthenticationManager authenticationManager = authenticationManager(http.getSharedObject(AuthenticationConfiguration.class));
-    HttpSecurity httpSecurity = http.authorizeRequests()
-            .antMatchers(HttpMethod.OPTIONS, "/**").
-            permitAll().
-            and().
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    HttpSecurity httpSecurity = http.authorizeRequests().
+            requestMatchers(EndpointRequest.to(
+                    InfoEndpoint.class,
+                    HealthEndpoint.class
+            )).permitAll().
+            requestMatchers(EndpointRequest.toAnyEndpoint()). //
+            hasAnyRole("ADMIN", "ACTUATOR"). //
+            antMatchers(HttpMethod.OPTIONS, "/**").
+            permitAll().and().
             sessionManagement().
-            sessionCreationPolicy(SessionCreationPolicy.STATELESS).and();
+            sessionCreationPolicy(SessionCreationPolicy.STATELESS).
+            and()
+            .csrf().disable();
```

Comment:
Keeping this, CSRF can never be enabled. So why keeping the property check afterwards?

Reply:
Just a C&P error. :-(